synapseclient/client.py

```diff
@@ -5247,3 +5270,146 @@ def _return_rest_body(self, response):
         if is_json(response.headers.get("content-type", None)):
             return response.json()
         return response.text
+
+
+@dataclass
+class Permissions:
```

Comment:
Let's move this out to a `models` directory to limit the amount of new content that we add to the `client.py`.

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/SYNPY-1322-OOP-POC/synapseclient/models/__init__.py is an example of what I had done for the Object-Oritentated-Programming PoC.

We could add this `Permissions` dataclass to both:
`synapseclient/models/__init__.py`
and
`synapseclient/__init__.py`

To allow either import.